lib/trace/trace_parser_loop.cpp

```diff
@@ -87,34 +79,23 @@ LastFrameLoopParser::parse_call(void)
 
     /* Restart last frame when looping is requested. */
     if (call) {
-        if (call->flags & trace::CALL_FLAG_END_FRAME) {
-            lastFrameBegin = lastFrameEnd;
-            parser->getBookmark(lastFrameEnd);
+        if (starts_new_frame) {
+            for (auto c : lastFrameCalls)
+                delete c;
+            lastFrameCalls.clear();
         }
+
+        call->reuse_call = true;
+        lastFrameCalls.push_back(call);
+        starts_new_frame = call->flags & trace::CALL_FLAG_END_FRAME;
+
+        lastFrameIterator = lastFrameCalls.begin();
+
         return call;
     }
 
     /* else */
     if (loopCount) {
```

Comment:
Given that now we have all the calls in the last frame available here, we might want to add a filtering step that removes all the calls that create or delete objects and are not paired. Creating more objects might crash the driver if the loop count is high. and destroying something might leave us with objects missing when the frame is replayed. But maybe that's for another PR.

Reply:
Good idea!

Reply:
The *not destroying* part is actually a bit more difficult, because at this point the destroy commands are already run before entering the loop.

Reply:
Will keep this in mind as we do more testing with this, thanks!